system/lib/fetch/emscripten_fetch.cpp

```diff
@@ -221,11 +221,15 @@ EMSCRIPTEN_RESULT emscripten_fetch_close(emscripten_fetch_t *fetch)
 	// which has not been yet closed. (double close is an error)
 	if (fetch->id == 0 || fetch->readyState > 4) return EMSCRIPTEN_RESULT_INVALID_PARAM;
 
+	unsigned short readyState = fetch->readyState;
+	fetch->readyState = 4; // DONE
+
 	// This fetch is aborted. Call the error handler if the fetch was still in progress and was canceled in flight.
-	if (fetch->readyState != 4 /*DONE*/ && fetch->__attributes.onerror)
+	if (readyState != 4 /*DONE*/ && fetch->__attributes.onerror)
 	{
 		fetch->status = (unsigned short)-1;
 		strcpy(fetch->statusText, "aborted with emscripten_fetch_close()");
+
 		fetch->__attributes.onerror(fetch);
```

Comment:
`fetch->__attributes` is an internal field so we can mutate that as well - perhaps assign `fetch->__attributes.onerror = fetch->__attributes.onsuccess = fetch->__attributes.onprogress = 0;` each so that those won't be called any more after the close?

Reply:
The whole fetch struct gets `free()`'d right away, hence that wouldn't have any impact, unless the structure is being accessed after free, as in scenario described in https://github.com/emscripten-core/emscripten/issues/8234.

Reply:
Another issue is that the `on*` handlers are captures in JS, so even clearing them is not a guarantee of not calling any other handlers.